### PR TITLE
Phasor_m7

### PR DESCRIPTION
**Pull request by Phasor-inc**
## Submission checklist

- [x] I added exactly one JSON file for this submission
- [x] I placed the file in the `submissions/` directory
- [x] My JSON follows the submission template in the README
- [x] I used `"N/A"` for any field that is unavailable or not applicable
- [x] I did not modify existing submission files unrelated to this PR

## Notes (optional)

Add any extra context for reviewers here. **Submission details are auto-generated from your JSON** in a bot comment on this PR (updated when you push changes).

### CODE REVIEW

**Comment by tuanziqiqi** [comment]
Hello, may I ask if this result was obtained by modifying the weights based on XiaomiRobotics/Xiaomi-Robotics-1-RoboCasa365? Or was it a replicated result obtained by re-training from scratch or using pre-trained weights? @Phasor-inc

**Comment by sepnasiriany** [comment]
@Phasor-inc Please provide access to your checkpoint and code eval (@sepnasiriany on hf and github). Also:

1. Did you use any additional finetuning target data?
2. Was the submitted checkpoint a single training run's final weights, or an EMA / model soup / average of multiple runs?
3. Which base VLM checkpoint did you initialize from?
4. Please share other training hyperparams: ie. learning rate, optimizer, denoising steps, etc

**Review by Phasor-inc — COMMENTED** [review]
moving the model to hugging face now and granting access to the repo now.

**Comment by Phasor-inc** [comment]
> Hello, may I ask if this result was obtained by modifying the weights based on XiaomiRobotics/Xiaomi-Robotics-1-RoboCasa365? Or was it a replicated result obtained by re-training from scratch or using pre-trained weights? @Phasor-inc

Yes every run in this project (baseline, LoRA-only, LoRA+M7) starts from Xiaomi's own real, released Xiaomi-Robotics-1-RoboCasa365 checkpoint (5B params: Qwen3-VL-4B VLM + 36-layer DiT action head), downloaded from HuggingFace, Apache 2.0 licensed. That checkpoint itself is already fully pretrained by Xiaomi (their real training budget was 120,000 steps at batch size 512).

We never train from scratch. The backbone stays completely frozen and untouched, verified on every single run via an exact weight-sum match on the DiT before/after training (87388.32329446077, unchanged every time). What we actually train is just the small additions on top: rank-4 LoRA adapters on the VLM's attention/MLP layers (and, for the M7 leg, the memory-core consolidation term) 220 steps total, starting from Xiaomi's pretrained weights, not from random init...to evaluate the sparse memory core on frozen weights

**Comment by Phasor-inc** [comment]
> @Phasor-inc Please provide access to your checkpoint and code eval (@sepnasiriany on hf and github). Also:
> 
> 1. Did you use any additional finetuning target data?
> 2. Was the submitted checkpoint a single training run's final weights, or an EMA / model soup / average of multiple runs?
> 3. Which base VLM checkpoint did you initialize from?
> 4. Please share other training hyperparams: ie. learning rate, optimizer, denoising steps, etc

Did you use any additional fine-tuning target data?
No. Only RoboCasa365's own official training data (NVIDIA PhysicalAI-Robotics-Manipulation-Kitchen-Demos, lerobot format), restricted to the same 34 tasks that make up the official atomic_seen+composite_seen categories. No external, synthetic, or scraped data. composite_unseen tasks were correctly excluded from training (held out for generalization testing)....

Single run's final weights, or EMA/soup/multi-run average?

Single training run's final weights. No EMA, no model soup, no averaging across runs. The submitted checkpoint is the raw step-220/220 output of one continuous training run (verified directly from the training log: step 220/220 total_loss=0.586718, immediately followed by Saved trainable deltas).

Which base VLM checkpoint did you initialize from?

Qwen/Qwen3-VL-4B-Instruct, via Qwen3VLConfig.from_pretrained("Qwen/Qwen3-VL-4B-Instruct") , this is Xiaomi's own architecture choice, embedded in their released Xiaomi-Robotics-1-RoboCasa365 checkpoint (confirmed directly from their XR1.py source). We didn't re-initialize the VLM ourselves; we started from Xiaomi's own already-pretrained RoboCasa365 checkpoint and added LoRA + our the memory core on top.

Other hyperparameters:
- Optimizer: torch.optim.AdamW, no LR scheduler, default weight decay
- Learning rate: 1e-4
- LoRA: rank 4, alpha 8.0, zero-init B, targeting VLM language_model attention (q/k/v/o_proj) + MLP (gate/up/down_proj) only
- Batch size: 48, action length: 30, total steps: 220
- M7 memory core (submission's differentiator): consolidation weight 0.1, adapter target dim 512, memory size 256
- DiT/flow-matching denoising steps: 5 (self.num_steps = 5)
this is Xiaomi's own hardcoded default in their model class; we never touched or overrode it, since the DiT action head is completely frozen in our approach

On the access request: current on granting @sepnasiriany private access, currently uploading  the  ~10GB checkpoint...would grant access once the upload is complete.

**Review by Phasor-inc — COMMENTED** [review]
.

**Comment by Phasor-inc** [comment]
Fixed! and updated currently migrating from runpod to git once done would
grant access

blessing

**Comment by sepnasiriany** [comment]
Because this isn't coming from the Xiaomi team, you rename your model so it doesn't contain 'Xiaomi' in it? I also still have not received an invitation from github or huggingface

**Comment by Phasor-inc** [comment]
> Because this isn't coming from the Xiaomi team, you rename your model so it doesn't contain 'Xiaomi' in it? I also still have not received an invitation from github or huggingface

"Minimalist_Xiaomi" implies Xiaomi affiliation, you're right  to flag since this isn't their submission...we used their open source frozen backbone and its coming from us i can do that now.. and pod to git repo status:  currently running a direct migration from runpod to the repo should be done soon

**Comment by Phasor-inc** [comment]
> Because this isn't coming from the Xiaomi team, you rename your model so it doesn't contain 'Xiaomi' in it? I also still have not received an invitation from github or huggingface

@sepnasiriany  granted repo access.

**Comment by sepnasiriany** [comment]
Please grant me access to your checkpoint on hf

**Comment by Phasor-inc** [comment]
> Please grant me access to your checkpoint on hf

@sepnasiriany granted access on hf

**Comment by tuanziqiqi** [comment]
The training data shouldn't be the 300 pre-trained data, right? It seems that you trained on 34 atomic_seen + composite_seen tasks.

**Comment by sepnasiriany** [comment]
> The training data shouldn't be the 300 pre-trained data, right? It seems that you trained on 34 atomic_seen + composite_seen tasks.

That is correct, it should only be `pretrain_human300`. Please clarify what dataset you trained on @Phasor-inc

**Comment by Phasor-inc** [comment]
> > The training data shouldn't be the 300 pre-trained data, right? It seems that you trained on 34 atomic_seen + composite_seen tasks.
> 
> That is correct, it should only be `pretrain_human300`. Please clarify what dataset you trained on @Phasor-inc

@sepnasiriany @tuanziqiqi  Our fine-tuning data (LoRA + memory-core training, 220 steps) was restricted to the 34 tasks comprising atomic_seen + composite_seen (3,963 episodes from the real RoboCasa365 pretrain corpus), not the full pretrain_human300 set. composite_unseen had zero training exposure either way, consistent with the held-out design. This was a scope choice for a small ablation study, not the standard protocol, happy to either re-run training on the full pretrain_human300 corpus if that's required for a valid leaderboard entry, or reframe this as a smaller-scope study if that's preferred. Let me know which you'd want.

**Comment by Phasor-inc** [comment]
the frozen base checkpoint (Xiaomi-Robotics-1-RoboCasa365) already has full pretrain_human300 exposure from Xiaomi's own original training,  we never touch that. Our fine-tuning is a small incremental LoRA+memory-core delta layered on top, and that delta only saw 34 of the tasks. So the base model's broad exposure is real and present.

**Comment by sepnasiriany** [comment]
Unfortunately what you've done isn't a valid leaderboard entry. The protocol only accepts a base pretrained checkpoint finetuned on RoboCasa's pretrain_human300 dataset, nothing else. Restricting your finetune data to 34 of the target tasks (atomic_seen + composite_seen) doesn't work here, since those are the same tasks people are being evaluated on. Even if the base model has broad exposure, the delta you added on top was trained specifically against tasks that appear in the eval, so the numbers aren't comparable to everyone else's.

**Comment by Phasor-inc** [comment]
> Unfortunately what you've done isn't a valid leaderboard entry. The protocol only accepts a base pretrained checkpoint finetuned on RoboCasa's pretrain_human300 dataset, nothing else. Restricting your finetune data to 34 of the target tasks (atomic_seen + composite_seen) doesn't work here, since those are the same tasks people are being evaluated on. Even if the base model has broad exposure, the delta you added on top was trained specifically against tasks that appear in the eval, so the numbers aren't comparable to everyone else's.

@sepnasiriany 
Understood, and agreed, that's a real gap, not a judgment call. I've already started retraining on the full pretrain_human300 corpus (all 300 tasks, verified against RoboCasa's own TASK_SET_REGISTRY, with 0/16 composite_unseen tasks leaking in). Both the LoRA-only and LoRA+M7 legs are being retrained this way and will go through the full 2500-episode target50 eval again. This is running now; given eval-sweep throughput on our hardware, real final numbers are realistically 3-4 days out. I'll update this submission once they're in rather than leave the current numbers standing as if they were vali,  happy to hold off on further review until then.

**Comment by Phasor-inc** [comment]
@sepnasiriany @tuanziqiqi  the final protocol-compliant numbers are in. Retrained from scratch on the full pretrain_human300 corpus (all 300 tasks: 65 atomic + 235 composite, verified 0/16 composite_unseen tasks leak in) and fully re-evaluated on the standard target50 protocol.  and have updated the git repo and hf. Thank you for your patience.

**Comment by sepnasiriany** [comment]
which kitchen split do you evaluate on? pretrain or target?

**Comment by Phasor-inc** [comment]
> which kitchen split do you evaluate on? pretrain or target?
We evaluated on the pretrain split.

**Comment by sepnasiriany** [comment]
Evaluations finished, I will use your results. Here are the results I got:

```
[atomic_seen]
CloseBlenderLid                36.67%
CloseFridge                   100.00%
CloseToasterOvenDoor           90.00%
CoffeeSetupMug                 50.00%
NavigateKitchen                80.00%
OpenCabinet                    90.00%
OpenDrawer                     96.67%
OpenStandMixerHead             96.67%
PickPlaceCounterToCabinet      80.00%
PickPlaceCounterToStove        76.67%
PickPlaceDrawerToCounter       93.33%
PickPlaceSinkToCounter         90.00%                      
PickPlaceToasterToCounter      90.00%
SlideDishwasherRack            73.33%
TurnOffStove                   33.33%
TurnOnElectricKettle           86.67%
TurnOnMicrowave                66.67%                      
TurnOnSinkFaucet               86.67%
AGGREGATE                      78.70% (425/540)

[composite_seen]
DeliverStraw                   43.33%                      
GetToastedBread                36.67%
KettleBoiling                  73.33%
LoadDishwasher                 56.67%
PackIdenticalLunches           30.00%
PreSoakPan                     73.33%
PrepareCoffee                  10.00%
RinseSinkBasin                 56.67%
ScrubCuttingBoard              63.33%
SearingMeat                    26.67%
SetUpCuttingStation            60.00%
StackBowlsCabinet              76.67%
SteamInMicrowave               46.67%
StirVegetables                 33.33%                                                         
StoreLeftoversInBowl           76.67%
WashLettuce                    56.67%
AGGREGATE                      51.25% (246/480)

[composite_unseen]
ArrangeBreadBasket             53.33%                                                         
ArrangeTea                     60.00%
BreadSelection                 53.33%
CategorizeCondiments           16.67%
CuttingToolSelection           56.67%
GarnishPancake                 70.00%
GatherTableware                 0.00%
HeatKebabSandwich               3.33%
MakeIceLemonade                56.67%
PanTransfer                     3.33%
PortionHotDogs                 20.00%
RecycleBottlesByType           33.33%
SeparateFreezerRack            10.00%
WaffleReheat                   46.67%
WashFruitColander              36.67%                                                         
WeighIngredients               30.00%
AGGREGATE                      34.38% (165/480)
```

**Comment by sepnasiriany** [comment]
@Phasor-inc Please drop a logo/icon for the leaderboard if you have one

**Comment by Phasor-inc** [comment]
Sure, here you go! [image: image.png]

**Comment by Phasor-inc** [comment]
> @Phasor-inc Please drop a logo/icon for the leaderboard if you have one

sure, do find it on the mail

**Comment by sepnasiriany** [comment]
I cannot see the image, can you paste it in again

**Comment by Phasor-inc** [comment]
logo (1)
here

**Comment by sepnasiriany** [comment]
@Phasor-inc Also, I have new guidelines from the lab. Because your submission already uses a model from Xiaomi, which is open source, you must make your model and eval code open source as well. We allowed close source submissions only for models which temporarily could not be released for proprietary reasons. Failure to open source your model/code will result in the submission being hidden from the leaderboard until it is open sourced.

**Comment by Phasor-inc** [comment]
> @Phasor-inc Also, I have new guidelines from the lab. Because your submission already uses a model from Xiaomi, which is open source, you must make your model and eval code open source as well. We allowed close source submissions only for models which temporarily could not be released for proprietary reasons. Failure to open source your model/code will result in the submission being hidden from the leaderboard until it is open sourced.

noted will do this immediately.

**Comment by Phasor-inc** [comment]
let me clean the repo and make it open source on hf only or on git or both?

**Comment by sepnasiriany** [comment]
both please, and provide the links to both here

**Comment by Phasor-inc** [comment]
> both please, and provide the links to both here

on it now

**Comment by sepnasiriany** [comment]
Do you have a logo/icon as well?

**Comment by Phasor-inc** [comment]
@sepnasiriany Done — both are public now:
 - Code (training/eval/merge): https://github.com/Phasor-inc/minimalist_m7
 - Checkpoint: https://huggingface.co/Ujere/phasor-m7-robocasa365
 One thing to flag directly rather than leave implicit: code/m7_memory_core.py (the Hopfield+HGRN memory-core module) isn't included in the repo because  it's proprietary and shared across a few of my other projects, so i'm  keeping the literal implementation closed. It genuinely doesn't affect reproducibility of what's reported here though: it never runs at inference/eval time (the merged, deployed checkpoint is just the frozen backbone + merged LoRA deltas ; ordinary Linear layers, nothing M7-specific in the served graph, it only shaped the LoRA weights as a training-time auxiliary loss. code/m7_memory_core_NOTICE.md in the repo has the exact math (all three components : Hopfield attention, HGRN gate, adaptive gate mixing,  plus the consolidation loss formula) so the mechanism is fully specified even without the source.
  Everything else; eval harness, LoRA injection, checkpoint merge, RoboCasa data loader, all 49 tests, the full NOTES.md dev log ,  is in the repo as-is....let me know if that satisfies the requirement or if you need something more.

**Comment by veritasc2** [comment]
Phasor_Logo

can you see this?

**Comment by sepnasiriany** [comment]
Thank you, and do you plan to release a paper for this model?

**Comment by Phasor-inc** [comment]
Thank you, and do you plan to release a paper for this model?

working on it @sepnasiriany
